contracts/governance/vote/adapter/INVVoteAdapterCharlieV1.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.3;
+
+import "OZ4/access/Ownable.sol";
+
+import "./IFixedRateAdapter.sol";
+import "../../token/IInverseToken.sol";
+
+/**
+ * @title Inverse vote adapter CharlieV1 version
+ * @notice This CharlieV1 version calculates the number of votes by getting a
+ * user's balance and multiplying that number by a fixed weight.
+ */
+contract INVVoteAdapterCharlieV1 is IFixedRateAdapter, Ownable {
+    /// @notice Address of the `INV` or `xINV` token.
+    IInverseToken public token;
+
+    /// @notice The voting weight associated with one `INV` or `xINV` token.
+    uint256 public weight;
+
+    /**
+     * @param token_ Address of the associated Inverse token.
+     * @param weight_ Initial weight associated with the token.
+     */
+    constructor(IInverseToken token_, uint256 weight_) Ownable() {
+        token = token_;
+        weight = weight_;
```

Comment:
would you deem it necessary to enforce an acceptable weight range?

Reply:
Not right off the bat. Perhaps something to look into later (as we gather info and analytics about Charlie on mainnet), we can easily replace a vote adapter through governance.